### ISSUE TITLE
Any great-master tell me why the authors (even alpha-fold-2 authors) use quaternion for rotation representation? is my understanding right?

### ISSUE THREAD

**Issue by yuedajiong** [graphdeco-inria/gaussian-splatting#448]
1. if we use raw N*N matrix to represent rotation, not only the data size is bigger than quaternion, but also:
    for rigid-body transformation, we need guarantee the matrix is orthogonal: R R^t=I
    how to guarantee?
    a) design the regular/loss, not very good;
    b) use svd in network, like this:
                U, S, Vt = torch.linalg.svd(R_soso)
                correction = torch.diag(torch.tensor([1, 1, torch.sign(torch.det(R_soso))], device=R_soso.device))
                R_good= (U @ correction) @ Vt
    c) directly use quaternion/Q4 as parameter, and this will avoid svd.   (same trick in alpha-fold-2)
        when we need the raw R, we can convert Q to R.
        for unit Q constraint, not simpler math idea, so used normlize_2 to guarantee.

2. not only rotation matrix vs. quaternion, but also others parameters like eq.6:  Sigma = R S S^t R^t
   the authors used similar mathematical method to guarantee correctness.

right?

**Comment by aruiplex** [comment]
You may refer this repo to understand: https://github.com/Krasjet/quaternion
